File: src/main/java/compiler/block/structures/FileBlock.java
package compiler.block.structures;

import compiler.block.Block;

/**
 * Represents the whole file.
 */
public class FileBlock extends Block {

    private String name;

    public FileBlock() {
        super(null, true, false);
    }

    public String getName() {
        return name;
    }

    @Override
    public String getValue() {
        return null;
    }

    @Override
    public String getType() {
        return null;
    }

    @Override
    public String getOpeningCode() {
        return "";
    }

    @Override
    public String getClosingCode() {
        return "";
    }

    @Override
    public String getBodyCode() {
        return "";
    }


    @Override
    public void init() {

    }

}

File: src/main/scala/test_classes/block/structures/FileBlock.scala
package test_classes.block.structures

import test_classes.block.Block

/**
  * Represents the whole file.
  */
class FileBlock(name: String) extends Block(null, true, false) {

  def init() {}

  def getName: String = name

  def getValue: String = ""

  def getType: String = "file"

  def getOpeningCode: String = ""

  def getClosingCode: String = ""

  override def toString: String = "file: " + name
}